### PR TITLE
SONARKT-36 Improve detection of 'this' inside of 'with', 'apply' and 'run' scope functions

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**

### CODE REVIEW

**Comment by johann-beleites-sonarsource** [comment]
Note that the low coverage is due to the chained null-safe calls, which are infeasible to cover without mocking.

File: sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/verifier/KotlinVerifier.kt [hunk 1]
```diff
@@ -48,7 +48,7 @@ private val DEFAULT_TEST_JARS_DIRECTORY = "../kotlin-checks-test-sources/build/t
 class KotlinVerifier(private val check: AbstractCheck) {
 
     var fileName: String = ""
-    var classpath: List<String> = System.getProperty("java.class.path").split(":") + DEFAULT_KOTLIN_CLASSPATH
+    var classpath: List<String> = System.getProperty("java.class.path").split(System.getProperty("path.separator")) + DEFAULT_KOTLIN_CLASSPATH
```
Comment (dorian-burihabwa-sonarsource) [inline sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/verifier/KotlinVerifier.kt:51]: Is this for portability? Or does it help this specific PR? Anyway, it is probably a good change

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM 👍🏿 (BTW I am surprised such a change does not affect the ruling)

File: sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/verifier/KotlinVerifier.kt (hunk 1 above)
Comment (johann-beleites-sonarsource) [reply sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/verifier/KotlinVerifier.kt:51]: Yea, it is for portability (on Windows, splitting by `:` is actually completely breaking the classpath - you need to use `;` instead). To be honest, I am surprised our unit tests all passed on Windows so far :thinking:

**Pull request merged**